commands-v2/ramsete/physics.py

```diff
@@ -0,0 +1,96 @@
+#
+# See the documentation for more details on how this works
+#
+# The idea here is you provide a simulation object that overrides specific
+# pieces of WPILib, and modifies motors/sensors accordingly depending on the
+# state of the simulation. An example of this would be measuring a motor
+# moving for a set period of time, and then changing a limit switch to turn
+# on after that period of time. This can help you do more complex simulations
+# of your robot code without too much extra effort.
+#
+
+from wpilib import RobotController, ADXRS450_Gyro
+from wpilib.simulation import (
+    PWMSim,
+    DifferentialDrivetrainSim,
+    EncoderSim,
+    ADXRS450_GyroSim,
+)
+from wpimath.system import LinearSystemId
+from wpimath.system.plant import DCMotor
+
+import constants
+
+from pyfrc.physics.core import PhysicsInterface
+
+
+class PhysicsEngine:
+    """
+    Simulates a motor moving something that strikes two limit switches,
+    one on each end of the track. Obviously, this is not particularly
+    realistic, but it's good enough to illustrate the point
+    """
+
+    def __init__(self, physics_controller: PhysicsInterface):
+
+        self.physics_controller = physics_controller
+
+        # Motor simulation definitions. Each correlates to a motor defined in
+        # the drivetrain subsystem.
+        self.m_frontLeftMotor = PWMSim(constants.kLeftMotor1Port)
```

Comment:
Can we remove the `m_` prefix on all the member variables?

Reply:
Will do.